File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        int n=nums.length;
        Map<Integer,Integer>map=new HashMap<>();
        for(int i=0;i<n;i++){
            map.put(nums[i],1);
        }
        ListNode newHead=null;
        ListNode temp=head;
        ListNode prev=null;
        while(temp!=null){
            int val=temp.val;
            if(!map.containsKey(val)){
                if(newHead==null){
                    newHead=temp;
                    prev=temp;
                }
                else{
                    prev.next=temp;
                    prev=prev.next;
                }
            }
            temp=temp.next;
        }
        if(prev!=null){
            prev.next=null;
        }
        return newHead;
    }
}

File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        unordered_set<int>set;

        for(auto num : nums)set.insert(num);

        ListNode* temp=head;
        ListNode* newHead=NULL;
        ListNode* prev=NULL;

        while(temp){
            int val=temp->val;
            if(set.find(val)==set.end()){
                if(!newHead){
                    newHead=temp;
                    prev=newHead;
                }
                else{
                    prev->next=temp;
                    prev=prev->next;
                }
            }
            temp=temp->next;
        }

        if(prev){
            prev->next=NULL;
        }

        return newHead;
    }
};